ehrapy/anndata/_feature_specifications.py

```
@@ -4,15 +4,75 @@
 import numpy as np
 import pandas as pd
 from anndata import AnnData
+from dateutil.parser import isoparse  # type: ignore
 from lamin_utils import logger
 from rich import print
 from rich.tree import Tree
 
-from ehrapy.anndata._constants import CATEGORICAL_TAG, CONTINUOUS_TAG, DATE_TAG, FEATURE_TYPE_KEY
-from ehrapy.anndata.anndata_ext import anndata_to_df
+from ehrapy.anndata._constants import CATEGORICAL_TAG, DATE_TAG, FEATURE_TYPE_KEY, NUMERIC_TAG
 
 
-def infer_feature_types(adata: AnnData, layer: str | None = None, output: Literal["tree", "dataframe"] | None = "tree"):
+def _detect_feature_type(col: pd.Series, verbose: bool = True) -> str:
+    """Detect the feature type of a column in a pandas DataFrame.
+
+    Args:
+        col: A pandas Series representing a feature.
+        verbose: Whether to print warnings for uncertain feature types. Defaults to True.
+
+    Returns:
+        The detected feature type. One of 'date', 'categorical', or 'numeric'.
+    """
+    n_elements = len(col)
+    col = col.dropna()
+    if len(col) == 0:
```

if a feature has only nans, I think it should raise an error and fail :)